adr/2025_03_10-suggestions/diff-yjs-snapshots.md

```diff
@@ -0,0 +1,32 @@
+
+# Diff two Y.js snapshots
+
+Y.js keeps track of metadata around what changes have been made to a document in order to efficiently synchronize changes. So, to compare a document with it's previous version, you can leverage this metadata by taking a snapshot of the full history of a document, and compare the insertions and deletions that have been made between the two snapshots. Because they share history, you can figure out what has been modified from _this point_ in history compared to _that point_ in history. For clarity, let's call this a `y-diff` for now to make sure it is different than a full content diff.
+
```

Comment:
basically, everything below here is about how to display the changes right (i.e.: how to bring them to the editor-level)? that wasn't immediately clear to me